### ISSUE TITLE
Relation does not exists error when changing table name after linking a foreign key to a field. On initial table creation.

### ISSUE THREAD

**Issue by mountain-hiker** [supabase/supabase#5192]
# Bug report

## Describe the bug

When renaming a table after you have linked a foreign key on app.supabase.io, table creation fails. This is when you are creating a new table in the UI from the website.

## To Reproduce

1. When creating a new table on app.supabase.io If you create a table with name: "my_table"
2. Then add a foreign key to a column on that table lets say "profile_id" -> profile.id (with the link button)
3. Then change the name of that table to  "my_test" before you save
4. An error is shown saying "relation public.my_table does not exist"

## Expected behavior

I would expect the table to be created.

## System information

- OS: macOS
- Browser (if applies) safari

## Additional context

Basically the name change is probably not reflected in the sql statement that adds the foreign key.

**Comment by martinzarfl** [comment]
@mountain-hiker I tried to reproduce the issue locally and on a supabase hosted project without success. Can you confirm that the issue still persists for you?

I tried two ways when setting up the table 
1. Saving "my-table" without the foreign key and then proceed with steps 2.-4.
2. Performing steps 1.-4. in one go

**Comment by saltcod** [comment]
Also can't reproduce, closing.
